lib/src/http/utils.dart

```diff
@@ -248,7 +248,7 @@ Map<String, Object> queryToMap(String query, {Encoding? encoding}) {
       if (fields[key] is! List) {
         fields[key] = [fields[key]];
       }
-      final List currentFields = fields[key] as List<dynamic>;
+      final currentFields = fields[key]! as List<dynamic>;
```

Comment:
we can be sure that `fields[key]` is not null by the `fields.containsKey(key)` if statement above, in combination with the `if fields[key] is! List`